pkg/distributor/distributor.go

```diff
@@ -1246,6 +1269,19 @@ func (d *Distributor) prePushValidationMiddleware(next PushFunc) PushFunc {
 			validatedExemplars += len(ts.Exemplars)
 		}
 
+		if len(dedupedPerMetric) > 0 {
+			// Emit tracing span events for metrics with deduped samples.
+			spanLogger, _ := spanlogger.NewWithLogger(ctx, d.log, "Distributor.prePushValidationMiddleware")
+			for m, c := range dedupedPerMetric {
+				spanLogger.DebugLog(
+					"msg", "deduplicated samples/histograms with conflicting timestamps from write request",
```

Comment:
```suggestion
					"msg", "deduplicated samples/histograms with identical timestamp and value from write request",
```

Reply:
The validation code only checks for duplicated timestamps though. Isn't it more correct to reflect this detail?